### PR TITLE
Port GoogleTest to QuRT (Hexagon RTOS)

### PR DESCRIPTION
**Pull request by Maratyszcza**
This change makes possible to build and use Google Test framework on Hexagon DSPs

### CODE REVIEW

File: googletest/src/gtest_main.cc [hunk 1]
```diff
@@ -32,18 +32,26 @@
 #include "gtest/gtest.h"
 
 #if GTEST_OS_ESP8266 || GTEST_OS_ESP32
-#if GTEST_OS_ESP8266
```
Comment (derekmauro) [inline googletest/src/gtest_main.cc:35]: This change seems unrelated to the issue this CL is addressing.

File: googletest/include/gtest/internal/gtest-port.h [hunk 2]
```diff
@@ -385,7 +385,8 @@ typedef struct _RTL_CRITICAL_SECTION GTEST_CRITICAL_SECTION;
 // On Android, <regex.h> is only available starting with Gingerbread.
 #define GTEST_HAS_POSIX_RE (__ANDROID_API__ >= 9)
 #else
-#define GTEST_HAS_POSIX_RE (!GTEST_OS_WINDOWS && !GTEST_OS_XTENSA)
+#define GTEST_HAS_POSIX_RE \
+  !(GTEST_OS_WINDOWS || GTEST_OS_XTENSA || GTEST_OS_QURT || GTEST_OS_QURT)
```
Comment (derekmauro) [inline googletest/include/gtest/internal/gtest-port.h:389]: It looks like GTEST_OS_QURT was added twice here.

**Review by derekmauro — CHANGES_REQUESTED** [review]

File: googletest/src/gtest_main.cc (hunk 1 above)
Comment (Maratyszcza) [reply googletest/src/gtest_main.cc:35]: Reverted
